airflow/providers/amazon/aws/hooks/batch_waiters.py

```diff
@@ -0,0 +1,210 @@
+        # of waiter_config leaking into the waiter_model.
+        waiters.waiter_model  # -> botocore.waiter.WaiterModel object
+
+        # the waiter_model is combined with the waiters.client to get a specific waiter
+        # and the details of the config on that waiter can be further modified without any
+        # accidental impact on the generation of new waiters from the defined waiter_model, e.g.
+        waiters.get_waiter("JobExists").config.delay  # -> 5
+        waiter = waiters.get_waiter("JobExists")  # -> botocore.waiter.Batch.Waiter.JobExists object
+        waiter.config.delay = 10
+        waiters.get_waiter("JobExists").config.delay  # -> 5  - defined by waiter_model
+
+        # to use a specific waiter, update the config and call the `wait()` method for jobId, e.g.
+        waiter = waiters.get_waiter("JobExists")  # -> botocore.waiter.Batch.Waiter.JobExists object
+        waiter.config.delay = randint(1, 10)  # seconds
+        waiter.config.max_attempts = 10
+        waiter.wait(jobs=[jobId])
+
+    .. seealso::
+
+        - https://www.2ndwatch.com/blog/use-waiters-boto3-write/
+        - https://github.com/boto/botocore/blob/develop/botocore/waiter.py
+        - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#waiters
+        - https://github.com/boto/botocore/tree/develop/botocore/data/ec2/2016-11-15
+        - https://github.com/boto/botocore/issues/1915
+    """
+
+    def __init__(self, waiter_config=None, aws_conn_id=None, region_name=None):
+
+        AwsBatchClient.__init__(self, aws_conn_id=aws_conn_id, region_name=region_name)
+
+        self._default_config = None
+        self._waiter_config = waiter_config or self.default_config
+        self._waiter_model = botocore.waiter.WaiterModel(self._waiter_config)
+
+    @property
+    def default_config(self) -> dict:
+        """An immutable default waiter configuration
+        :return: a waiter configuration for AWS batch services
+        :rtype: dict
+        """
+        if self._default_config is None:
+            config_path = Path(__file__).with_name("batch_waiters.json").absolute()
+            with open(config_path) as config_file:
+                self._default_config = json.load(config_file)
+        # return a deep copy, to avoid accidental mutation
+        return deepcopy(self._default_config)
+
+    @property
+    def waiter_config(self):
+        """
+        :return: a mutable waiter configuration for AWS batch services
+        :rtype: dict
+        """
+        return self._waiter_config
+
+    @property
+    def waiter_model(self):
+        """
+        :return: a waiter model for AWS batch services
+        :rtype: botocore.waiter.WaiterModel
+        """
+        return self._waiter_model
+
+    def get_waiter(self, waiter_name):
+        """
+        Get an AWS Batch service waiter
+
+        :param waiter_name: The name of the waiter.  The name should match
```

Comment:
```suggestion
        :param waiter_name: The name of the waiter. The name should match
```

Reply:
OK, this is likely a remnant of my academic education in writing science articles.